src/check_python_versions/sources/pyproject.py

```diff
@@ -0,0 +1,364 @@
+"""
+Support for pyproject.toml.
+
+There are two ways of declaring Python versions in a pyproject.toml:
+classifiers like
+
+    Programming Language :: Python :: 3.8
+
+and tool.poetry.dependencies.python keyword.
+
+check-python-versions supports both.
```

Comment:
I think it's worth mentioning that there are multiple different build systems that use pyproject.toml, and check-python-versions supports the following ones:

- poetry
- flit
- setuptools

(judging from the names of the unit test files).  A brief mention of which settings these actually are (tool.poetry.classifiers, tool.poetry.dependencies.python) for each would be a great plus.  Currently I don't think we need to go down into actual examples of what the classifiers look like.